system-addon/lib/ManualMigration.jsm

```diff
@@ -23,20 +25,22 @@ this.ManualMigration = class ManualMigration {
     Services.obs.removeObserver(this, MIGRATION_ENDED_EVENT);
   }
 
-  isMigrationMessageExpired() {
+  async isMigrationMessageExpired() {
     let migrationLastShownDate = new Date(this._prefs.get("migrationLastShownDate") * 1000);
     let today = new Date();
     // Round down to midnight.
     today = new Date(today.getFullYear(), today.getMonth(), today.getDate());
-
     if (migrationLastShownDate < today) {
+      let profileAge = new ProfileAge(null, null);
+      let profileCreationDate = await profileAge.created;
+      let daysSinceProfileCreation = (Date.now() - profileCreationDate) / MS_PER_DAY;
```

Comment:
The reason I had it like was to reduce the number of `return`s in the function. Makes it easier to reason (at least for me). But I don't feel strongly about it.